x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_settings_form/processors/common_fields/processor_type_field.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { EuiComboBox } from '@elastic/eui';
+import React, { FunctionComponent } from 'react';
+import {
+  FIELD_TYPES,
+  FieldConfig,
+  UseField,
+  fieldValidators,
+  FormRow,
+} from '../../../../../../../shared_imports';
+import { types } from '../../map_processor_type_to_form';
+
+interface Props {
+  initialType?: string;
+}
+
+const { emptyField } = fieldValidators;
+
+const typeConfig: FieldConfig = {
+  type: FIELD_TYPES.TEXT,
```

Comment:
IIRC there is a `FIELD_TYPES.COMBOBOX` that you can use.

Reply:
@alisonelizabeth Thanks for pointing that out? Do you know what effect setting it to `FIELD_TYPES.COMBO_BOX` has?

Reply:
It maps to this: https://github.com/elastic/kibana/blob/master/src/plugins/es_ui_shared/static/forms/components/fields/combobox_field.tsx

(Not tested) I _think_ you can do something like this:

```
const typeConfig: FieldConfig = {
  type: FIELD_TYPES.COMBO_BOX,
  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.typeField.typeFieldLabel', {
    defaultMessage: 'Type',
  }),
  defaultValue: [],
  validations: [
    {
      validator: emptyField(
        i18n.translate('xpack.ingestPipelines.pipelineEditor.typeField.fieldRequiredError', {
          defaultMessage: 'A type is required.',
        })
      ),
    },
  ],
};
```

```
<UseField config={typeConfig} path="type" />
```